### PR TITLE
Begin yarpc-python

### PR DESCRIPTION
**Pull request by breerly**
- [x] move in urpc code
- [x] configure travis
- [x] rename urpc to yarpc
- [x] rename scheme to encoding
- [x] add travis badge to readme.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+language: python
+
+python:
+    - 2.7
+    - pypy
+    - 3.3
+    - 3.4
+    - 3.5
+    - pypy3
```
Comment (abhinav) [inline .travis.yml]: heh nice

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu-debootstrap:trusty
```
Comment (abhinav) [inline Dockerfile]: You probably wanted to use alpine here? Although that can probably wait until a later PR.

**Comment by abhinav** [comment]
are you planning on doing the `s/urpc/yarpc` in this PR or later?

**Comment by breerly** [comment]
@abhinav look at the todo list in the PR

File: Dockerfile (hunk 2 above)
Comment (breerly) [inline Dockerfile]: I'm killing this file - this was for aurora

**Comment by breerly** [comment]
OK - this is ready to review. I made some issues to address (python3 support and tox) in order to start iterating.

cc @yarpc/python @junchaowu @abhinav @blampe

File: examples/async.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+parser.add_option('-i', '--inbound', dest='inbound',
+                  default='localhost:8888', help='inbound hostport')
+parser.add_option('-o', '--outbound', dest='outbound',
+                  default='localhost:8888', help='outbound hostport')
+(options, args) = parser.parse_args()
+
+inbound_port = int(options.inbound.split(':')[1])
+outbound_hostport = options.outbound
+
+# config urpc
```
Comment (breerly) [inline examples/async.py]: Change comment to yarpc.

File: examples/sync.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+from yarpc_sync.inbounds import HTTPInbound
+
+rpc = RPC(
+    service='test-service',
+    inbounds=[HTTPInbound(port=8888)],
+)
+
+
+def test(request):
+    # resp = urpc.json('billy', {'hello':'world'})
```
Comment (breerly) [inline examples/sync.py]: change comment to rpc

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+setup(
+    name='yarpc',
+    packages=find_packages(exclude=['tests', 'tests.*']),
+    install_requires=[
+
+        # async
+        'tornado>=4.2,<5',
+        'thriftrw>=1.1,<2',
+
+        # sync
```
Comment (breerly) [inline setup.py]: move to sync package... @abhinav @blampe you guys know how to seperate the setup.py for the 2 packages in this mega repo?

**Comment by breerly** [comment]
Add the following to readme:

```
[![Build Status](https://travis-ci.org/yarpc/yarpc-python.svg?branch=master)](https://travis-ci.org/yarpc/yarpc-python)
```

File: setup.py (hunk 5 above)
Comment (blampe) [inline setup.py]: You mean like you want people to install `yarpc[sync]`? That's done with the `extras_require` kwarg.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+sudo: required
+
+language: python
+
+python:
+    - 2.7
+    - pypy
```
Comment (blampe) [inline .travis.yml]: No 3?

**Comment by blampe** [comment]
Go for it.

File: README.md [hunk 7]
```diff
@@ -1 +1,2 @@
-# yarpc-python
\ No newline at end of file
+# yarpc-python:
+
```
Comment (junchaowu) [inline README.md]: the build status?

File: requirements.txt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+pytest>=2,<3
```
Comment (junchaowu) [inline requirements.txt]: should all these in requirements-test.txt?

File: setup.py (hunk 5 above)
Comment (abhinav) [inline setup.py]: or if you want to actually have two separate pypi packages, you need to actually create two directories with separate setup.py and package trees

**Comment by abhinav** [comment]
:shipit: 

I assume API updates will come in a separate PR.

File: .travis.yml (hunk 6 above)
Comment (breerly) [inline .travis.yml]: I created an issue so that can come in another PR - short answer: didn't pass.

File: requirements.txt (hunk 8 above)
Comment (breerly) [inline requirements.txt]: in a library, `requirements.txt` is basically the same thing

File: setup.py (hunk 5 above)
Comment (breerly) [inline setup.py]: got it - created #4 to track

**Comment by breerly** [comment]
@abhinav created #5 to track API updates.

**Pull request merged**